### DISCUSSION TITLE
Different root path for different upstream targets

### DISCUSSION THREAD

**Discussion by Zorlin** [Kong/kong#8882] [Help]
Hi there,

I'm trying out Kong as a load balancer for a number of RPC endpoints.

My situation is very similar to this: https://stackoverflow.com/questions/70817573/kong-load-balance-where-upstream-has-different-root-paths

I want to be able to POST a request to https://kong-server/ (at the root path), and have it routed to one of 3 targets.

I think what I'm after is to have an upstream with ~3 targets, but some of those targets need a specific root path (eg /v2/apikey/).

Is there a way to configure a target to have a different path than the other targets?

**Comment by jschmid1** [comment]
Just to clarify and confirm that I got it. You're trying to configure one route to root `/` that implements one service `service_x` which in turn has 3 distinct upstream servers that hold the same content and can be safely load-balanced, but have to be contacted using different paths (`/v1, /v2 /v3`).

**Comment by Zorlin** [comment]
Yep, exactly correct :)
